File: python/sorting_algorithms/notable_sorting_algorithms/gnomeSort.py
def gnome_sort(arr):
    i = 0
    while i < len(arr):
        if i == 0 or arr[i - 1] <= arr[i]:
            i += 1
        else:
            arr[i], arr[i - 1] = arr[i - 1], arr[i]
            i -= 1

# Ejemplo de uso
arr = [34, 2, 10, -9, 7]
print("\nArray original:", arr)
gnome_sort(arr)
print("\nArray ordenado:", arr)

File: c/sorting_algorithms/notable_sorting_algorithms/gnomeSort.c
#include <stdio.h>

// Gnome Sort
void gnomeSort(int arr[], int n) {
    int i = 0;
    while (i < n) {
        if (i == 0 || arr[i - 1] <= arr[i]) {
            i++;
        } else {
            int temp = arr[i];
            arr[i] = arr[i - 1];
            arr[i - 1] = temp;
            i--;
        }
    }
}

// Imprimir un array
void printArray(int arr[], int n) {
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

int main() {
    int arr[] = {34, 2, 10, -9, 7};
    int n = sizeof(arr) / sizeof(arr[0]);

    printf("\nArray original:\n");
    printArray(arr, n);

    gnomeSort(arr, n);

    printf("\nArray ordenado:\n");
    printArray(arr, n);

    return 0;
}